### PR TITLE
Discard negative values from pie chart post-aggregation instead of during summation

### LINKED ISSUE #5481: Individual negative values discarded in pie chart
It looks like negative values are dropped when constructing pie charts, probably to prevent negative sums ([example](https://codepen.io/justin-sleep/pen/dyOXJmx?editors=1010)). I think [this](https://github.com/plotly/plotly.js/blob/dc3c342f346bd6effad210414f377345fe3e2d47/src/traces/pie/calc.js#L42) is the line responsible for this behavior, and that it could be fixed by checking for negative values in the sums rather than checking the individual points.

### PR DESCRIPTION
**Pull request by destiny-wu**
Fixes #5481  by retaining negative values during summation and dropping negative totals afterward. Adds `pie_negative_values.json` to the test image mocks for demonstration of the changed behavior.

### CODE REVIEW

File: src/traces/pie/calc.js [hunk 1]
```diff
@@ -76,6 +75,9 @@ function calc(gd, trace) {
         }
     }
 
+    // Drop aggregate sums of value 0 or less
+    cd = cd.filter(function(elem) { return elem.v > 0; });
```
Comment (archmoj) [inline src/traces/pie/calc.js:79]:
```suggestion
    cd = cd.filter(function(elem) { return elem.v >= 0; });
```

Comment (archmoj) [reply src/traces/pie/calc.js:79]: So that `funnelarea_line_width` mock render as expected.

File: test/image/mocks/pie_negative_values.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          "domain": {"x": [0.5, 1]},
+          "textinfo": "label+value+percent"
+      }
+    ],
+    "layout": {
+      "height": 300,
+      "width": 500
+    }
+  }
+
```
Comment (archmoj) [inline test/image/mocks/pie_negative_values.json:42]: Please fix this file. `npm run test-syntax` complains about 
`test/image/mocks/pie_negative_values.json : does not have a trailing new line character`

Comment (archmoj) [reply test/image/mocks/pie_negative_values.json:42]: Please also add and commit this image in the `test/image/baselines/` folder.
Thanks!
pie_negative_values

File: src/traces/pie/calc.js (hunk 1 above)
Comment (destiny-wu) [reply src/traces/pie/calc.js]: Done

File: test/image/mocks/pie_negative_values.json (hunk 2 above)
Comment (destiny-wu) [reply test/image/mocks/pie_negative_values.json]: Added this image to `test/image/baselines/`

**Comment by archmoj** [comment]
Currently there is a problem in our image test system.
Would you please rename the `png` and `json` files to start with letter `z`?
`pie_negative_values` > `z-pie_negative_values`
Thanks!

**Comment by archmoj** [comment]
Thanks very much for the PR.
:dancer:

**Pull request merged**